swap_meet/vendor.py

```diff
@@ -1,2 +1,57 @@
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self,inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+
+        return [item for item in self.inventory if category == item.category]
+
+    def swap_items(self,friend, my_item, their_item):
+
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.inventory.remove(my_item)
+            friend.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            friend.inventory.append(my_item)
+            return True
+        return False
```

Comment:
We can use a guard clause here too to flip the logic so that we have a check to make sure we're working with valid data. If the data isn't valid we'll return false and then the meat of your logic can be un-indented in the lines that follow:

```python
if my_item not in self.inventory or their_item not in other_vendor.inventory:
    return False

# rest of your swapping logic here
```

Also, recall that you already wrote methods for the Vendor class that allow you to add and remove from the inventory (without needing to access the `inventory` attribute on each Vendor object and using the Python List methods remove and append). You can reuse those methods on lines 23-26:
```python
self.remove(my_item)
friend.remove(their_item)
self.add(their_item)
friend.add(my_item)